js/pages/tola_management_pages/country/components/edit_disaggregations.js

```diff
@@ -249,6 +331,13 @@ class DisaggregationType extends React.Component {
         }, () => this.hasUnsavedDataAction());
     }
 
+    updateRetroPrograms(id, checked) {
+        runInAction(() => {
+            this.programsForRetro[id]['checked'] = checked
+        })
+        this.hasUnsavedDataAction()
+    }
+
```

Comment:
just for pure code simplicity (since this is already pulled out to its own method) you could just tag this `@action` (from what I understand of `runInAction` it's when you don't want to pull an action out to its own method so you just instantiate a one-off in the middle of functional code, I think?